drivers/ethernet/eth_stm32_hal.c

```diff
@@ -1486,27 +1296,30 @@ static const struct eth_stm32_hal_dev_cfg eth0_config = {
 
 static struct eth_stm32_hal_dev_data eth0_data = {
 	.heth = {
-		.Instance = (ETH_TypeDef *)DT_INST_REG_ADDR(0),
-		.Init = {
-#if !defined(CONFIG_SOC_SERIES_STM32H7X) && !defined(CONFIG_SOC_SERIES_STM32H5X) && \
-	!defined(CONFIG_ETH_STM32_HAL_API_V2)
+			.Instance = (ETH_TypeDef *)DT_INST_REG_ADDR(0),
+			.Init = {
+#if !defined(CONFIG_ETH_STM32_HAL_API_V2)
 #if defined(CONFIG_ETH_STM32_AUTO_NEGOTIATION_ENABLE)
-			.AutoNegotiation = ETH_AUTONEGOTIATION_ENABLE,
+					.AutoNegotiation = ETH_AUTONEGOTIATION_ENABLE,
 #else
-			.AutoNegotiation = ETH_AUTONEGOTIATION_DISABLE,
-			.Speed = IS_ENABLED(CONFIG_ETH_STM32_SPEED_10M) ?
-				 ETH_SPEED_10M : ETH_SPEED_100M,
-			.DuplexMode = IS_ENABLED(CONFIG_ETH_STM32_MODE_HALFDUPLEX) ?
-				      ETH_MODE_HALFDUPLEX : ETH_MODE_FULLDUPLEX,
+					.AutoNegotiation = ETH_AUTONEGOTIATION_DISABLE,
+					.Speed = IS_ENABLED(CONFIG_ETH_STM32_SPEED_10M)
+							 ? ETH_SPEED_10M
+							 : ETH_SPEED_100M,
+					.DuplexMode = IS_ENABLED(CONFIG_ETH_STM32_MODE_HALFDUPLEX)
+							      ? ETH_MODE_HALFDUPLEX
+							      : ETH_MODE_FULLDUPLEX,
 #endif /* !CONFIG_ETH_STM32_AUTO_NEGOTIATION_ENABLE */
-			.PhyAddress = PHY_ADDR,
-			.RxMode = ETH_RXINTERRUPT_MODE,
-			.ChecksumMode = IS_ENABLED(CONFIG_ETH_STM32_HW_CHECKSUM) ?
-					ETH_CHECKSUM_BY_HARDWARE : ETH_CHECKSUM_BY_SOFTWARE,
-#endif /* !CONFIG_SOC_SERIES_STM32H7X */
-			.MediaInterface = IS_ENABLED(CONFIG_ETH_STM32_HAL_MII) ?
-					  ETH_MEDIA_INTERFACE_MII : ETH_MEDIA_INTERFACE_RMII,
-		},
+					.PhyAddress = PHY_ADDR,
+					.RxMode = ETH_RXINTERRUPT_MODE,
+					.ChecksumMode = IS_ENABLED(CONFIG_ETH_STM32_HW_CHECKSUM)
+								? ETH_CHECKSUM_BY_HARDWARE
+								: ETH_CHECKSUM_BY_SOFTWARE,
+#endif /* !CONFIG_ETH_STM32_HAL_API_V2 */
+					.MediaInterface = IS_ENABLED(CONFIG_ETH_STM32_HAL_MII)
+								  ? ETH_MEDIA_INTERFACE_MII
+								  : ETH_MEDIA_INTERFACE_RMII,
+			},
```

Comment:
I'm not sure about the indentation shift in this would section.
If really required, please make it in a dedicated commit.

Reply:
It's required in the compilation checks. I can remove it.
![Screenshot from 2024-10-18 12-01-54](https://github.com/user-attachments/assets/4cd5c9e7-6391-4ad8-9e91-8fc5f65b120b)

Reply:
Ok, these are just warnings which won't affect result warning, so not mandatory to fix.
In general case, if you chose to fix them, please do it in a dedicated commit.